library/init/category/lift.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2016 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner, Sebastian Ullrich
+-/
+prelude
+import init.function init.coe
+import init.category.monad
+import init.util
+
+universes u v w
+
+/-- A function for lifting a computation from an inner monad to an outer monad.
+    Like https://hackage.haskell.org/package/transformers-0.5.5.0/docs/Control-Monad-Trans-Class.html,
+    but `n` does not have to be a monad transformer. -/
+class has_monad_lift (m : Type u → Type v) (n : Type u → Type w) :=
+(monad_lift {} : ∀ {α}, m α → n α)
+
+/-- The reflexive-transitive closure of `has_monad_lift`. -/
+class has_monad_lift_t (m : Type u → Type v) (n : Type u → Type w) :=
+(monad_lift {} : ∀ {α}, m α → n α)
+
+export has_monad_lift_t (monad_lift)
+
+@[reducible] def has_monad_lift_to_has_coe {m n} [has_monad_lift_t m n] {α} : has_coe (m α) (n α) :=
+⟨monad_lift⟩
+
+instance has_monad_lift_t_trans (m n o) [has_monad_lift n o] [has_monad_lift_t m n] : has_monad_lift_t m o :=
+⟨λ α ma, has_monad_lift.monad_lift (monad_lift ma : n α)⟩
+
+instance has_monad_lift_t_refl (m) : has_monad_lift_t m m :=
+⟨λ α, id⟩
+
+@[simp] lemma monad_lift_refl {m : Type u → Type v} {α} : (monad_lift : m α → m α) = id := rfl
+
+
+/-- A functor in the category of monads. Can be used to lift monad-transforming functions.
+    Based on https://hackage.haskell.org/package/pipes-2.4.0/docs/Control-MFunctor.html,
+    but not restricted to monad transformers. -/
+class monad_functor (m m' : Type u → Type v) (n n' : Type u → Type w) :=
```

Comment:
Do you have plans for this `class`? Why do we need it?
I found very few occurrences in the rest of the PR.
I also did not find any good motivation for it in the link above.